host/esutils/client_os2.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package esutils
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	os2Client struct {
+		client *opensearch.Client
+	}
+)
+
+func newOS2Client(url string) (*os2Client, error) {
+
+	esClient, err := opensearch.NewClient(
+		opensearch.Config{
+			Addresses:    []string{url},
+			MaxRetries:   5,
+			RetryBackoff: func(i int) time.Duration { return time.Duration(i) * 100 * time.Millisecond },
+		},
+	)
+
+	return &os2Client{
+		client: esClient,
+	}, err
+}
+
+func (os2 *os2Client) PutIndexTemplate(t *testing.T, templateConfigFile, templateName string) {
+	// This function is used exclusively in tests. Excluding it from security checks.
+	// #nosec
+	template, err := os.Open(templateConfigFile)
+	require.NoError(t, err)
+
+	req := opensearchapi.IndicesPutIndexTemplateRequest{
+		Body: template,
+		Name: templateName,
+	}
+
+	resp, err := req.Do(createContext(), os2.client)
+	require.NoError(t, err)
+	require.False(t, resp.IsError(), fmt.Sprintf("OS2 error: %s", resp.String()))
+	resp.Body.Close()
+
+}
+
+func (os2 *os2Client) CreateIndex(t *testing.T, indexName string) {
+	existsReq := opensearchapi.IndicesExistsRequest{
+		Index: []string{indexName},
+	}
+	resp, err := existsReq.Do(createContext(), os2.client)
+	require.NoError(t, err)
+
+	if resp.StatusCode == http.StatusOK {
+		deleteReq := opensearchapi.IndicesDeleteRequest{
```

Comment:
kinda odd that these delete the index implicitly... but eh.  the others do too.

Reply:
yes, that seems to be a bit odd, given that `TearDownSuite` should delete that index anyway.